Exercise3/MainActivity.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Android.App;
+using Android.OS;
+using Android.Support.V7.Widget;
+using Com.Lilarcor.Cheeseknife;
+using Exercise3.Controllers;
+using Exercise3.Controllers.Adapters;
+using SearchView = Android.Widget.SearchView;
+
+namespace Exercise3
+{
+    [Activity(Label = "Photo Finder", MainLauncher = true, Theme = "@android:style/Theme.Holo.Light")]
+    public class MainActivity : Activity
+    {
+        [InjectView(Resource.Id.rvPhotos)] private RecyclerView rvPhotos;
+
+        private SearchView searchView;
+        private ImageAdapter adapter;
+        private GettyImageApiController apiController;
+        
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+
+            SetContentView(Resource.Layout.Main);
+            Cheeseknife.Inject(this);
+            rvPhotos.SetLayoutManager(new LinearLayoutManager(this));
+            apiController = new GettyImageApiController();
+
+            searchView = FindViewById<SearchView>(Resource.Id.searchView);
+            searchView.QueryTextSubmit += (sender, e) =>
+            {
+                adapter = new ImageAdapter(apiController.SearchImage(searchView.Query));
```

Comment:
You should update the existing adapter instead of creating a new one.

Reply:
i dont know why but when i code like:
image = apiController.SearchImage(searchView.Query)
adapter.NotifyDataSetChanged();

then it doesn't work :(

Reply:
Because `apiController.SearchImage(searchView.Query)` will create a new list and the adapter is holding another list.

Reply:
If my answer isn't enough detail for you can understand. You can change like this:
Step 1. Move this line to above
```C#
adapter = new ImageAdapter(); //Create an empty constructor or remove the existing one.
```

Step 2. Make `List<Image> images` to become a property
```C#
public List<Image> Images{
    set
    {
       this.images = value;
       this.NotifyDataseChanged();
    }
}
```

Step 3.
```C#
searchView.QueryTextSubmit += (sender, e) =>
{
  adapter.Images = apiController.SearchImage(searchView.Query);
}
```

It should work

Reply:
ok, got it, I'll try it.
Thanks you